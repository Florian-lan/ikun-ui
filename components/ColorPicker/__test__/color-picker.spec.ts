import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import KColorPicker from '../src';

let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.useRealTimers();
});

describe('Test: KColorPicker', () => {
	test('props: cls', async () => {
		const instance = new KColorPicker({
			target: host,
			props: {
				cls: 'k-color-picker--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-color-picker--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
