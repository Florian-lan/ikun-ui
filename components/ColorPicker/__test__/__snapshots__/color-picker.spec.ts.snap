// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KColorPicker > props: cls 1`] = `"<div aria-hidden="true" class="flex" data-popover-trigger=""><div slot="triggerEl" class=""><div class="k-color-picker-block-w k-color-picker-block-w--md k-color-picker-block-w--focus"><div class="k-color-picker-block k-color-picker-block--md k-color-picker-block--md"><div class="k-color-picker-block-content--md" style="background-color: rgb(0, 0, 0);"></div></div> <div slot="text" class="" style="display: none;"></div></div></div></div> <div class="k-popover--base k-popover--base__top k-popover--base__top__dark k-popover--base__dark svelte-13arsd0" data-popper-placement="top" aria-hidden="true" style="position: fixed; left: 0px; top: 0px; margin: 0px; animation: __svelte_3896445108_0 200ms linear 0ms 1 both; bottom: 8px;" data-popper-reference-hidden="" data-popper-escaped=""><div slot="contentEl" class="k-color-picker k-color-picker--test"><div class="k-color-picker-header"><span class=""></span> </div> <div><div class="k-color-picker-palette k-color-picker-palette--base" aria-hidden="true" style="background-color: rgb(0, 0, 0);"><div class="k-color-picker-pointer" aria-hidden="true" style="left: -99px; top: -99px;"></div></div> <div class="k-color-picker--hsb"><div class="k-color-picker--hs"><div aria-hidden="true" class="k-color-picker-slider"><div class="k-color-picker-slider--track"></div> <div class="k-color-picker-slider--thumb" style="background-color: rgb(255, 0, 0); left: 0%;"></div></div> <div aria-hidden="true" class="k-color-picker-slider k-color-picker-slider_alpha k-color-picker--alpha"><div class="k-color-picker-slider--track_alpha"></div> <div class="k-color-picker-slider--thumb" style="background-color: rgb(0, 0, 0); left: 100%;"></div></div></div> <div class="k-color-picker-block"><div class="k-color-picker-block-content" style="background-color: rgb(0, 0, 0);"></div></div></div> <div class="k-color-picker-format"><div aria-hidden="true" class="flex" data-popover-trigger=""><button class="k-dropdown" role="menu" tabindex="0" slot="triggerEl"><div class="k-color-picker-format--val">RGB <span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-down op50" role="" aria-hidden="true"></span></div></button></div>  <div class="k-input k-color-picker-format--input-num svelte-1ibt20v"> <div class="k-input--base k-input-number__sm k-input__dark k-input__hover k-input__focus k-input__rounded svelte-1ibt20v"><input type="number" class="k-input--inner k-input--inner__dark svelte-1ibt20v" min="0" max="255" placeholder="" name="" id=""> <div class="k-input-number--controls svelte-1ibt20v"><div class="k-input-number--up k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="decrease number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-up k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div> <div class="k-input-number--down k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="increase number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-down k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div></div></div> </div> <div class="k-input k-color-picker-format--input-num svelte-1ibt20v"> <div class="k-input--base k-input-number__sm k-input__dark k-input__hover k-input__focus k-input__rounded svelte-1ibt20v"><input type="number" class="k-input--inner k-input--inner__dark svelte-1ibt20v" min="0" max="255" placeholder="" name="" id=""> <div class="k-input-number--controls svelte-1ibt20v"><div class="k-input-number--up k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="decrease number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-up k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div> <div class="k-input-number--down k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="increase number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-down k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div></div></div> </div> <div class="k-input k-color-picker-format--input-num svelte-1ibt20v"> <div class="k-input--base k-input-number__sm k-input__dark k-input__hover k-input__focus k-input__rounded svelte-1ibt20v"><input type="number" class="k-input--inner k-input--inner__dark svelte-1ibt20v" min="0" max="255" placeholder="" name="" id=""> <div class="k-input-number--controls svelte-1ibt20v"><div class="k-input-number--up k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="decrease number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-up k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div> <div class="k-input-number--down k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="increase number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-down k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div></div></div> </div>  <div class="k-input k-color-picker-format--input-num svelte-1ibt20v"> <div class="k-input--base k-input-number__sm k-input__dark k-input__hover k-input__focus k-input__rounded svelte-1ibt20v"><input type="number" class="k-input--inner k-input--inner__dark svelte-1ibt20v" min="1" max="100" placeholder="" name="" id=""> <div class="k-input-number--controls svelte-1ibt20v"><div class="k-input-number--up k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="decrease number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-up k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div> <div class="k-input-number--down k-input-number--controls__dark svelte-1ibt20v" aria-hidden="true" role="button" aria-label="increase number"><span class="k-icon--base k-icon--base__dark k-icon-transition i-carbon-chevron-down k-input-number--icon k-input--icon__sm" role="" aria-hidden="true" style="width: 24px; height: 24px;"></span></div></div></div> </div></div> </div></div> </div>"`;

exports[`Test: KColorPicker > props: showText 1`] = `"<div aria-hidden="true" class="flex" data-popover-trigger=""><div slot="triggerEl" class=""><div class="k-color-picker-block-w k-color-picker-block-w--md"><div class="k-color-picker-block k-color-picker-block--md k-color-picker-block--md"><div class="k-color-picker-block-content--md" style="background-color: rgb(22, 119, 255);"></div></div> <div slot="text" class="k-color-picker-txt" style="display: initial;">#1677ffff</div></div></div></div> "`;

exports[`Test: KColorPicker > props: size 1`] = `"<div aria-hidden="true" class="flex" data-popover-trigger=""><div slot="triggerEl" class=""><div class="k-color-picker-block-w k-color-picker-block-w--sm"><div class="k-color-picker-block k-color-picker-block--sm k-color-picker-block--sm"><div class="k-color-picker-block-content--sm" style="background-color: rgb(0, 0, 0);"></div></div> <div slot="text" class="" style="display: none;"></div></div></div></div> <div aria-hidden="true" class="flex" data-popover-trigger=""><div slot="triggerEl" class=""><div class="k-color-picker-block-w k-color-picker-block-w--md"><div class="k-color-picker-block k-color-picker-block--md k-color-picker-block--md"><div class="k-color-picker-block-content--md" style="background-color: rgb(0, 0, 0);"></div></div> <div slot="text" class="" style="display: none;"></div></div></div></div> <div aria-hidden="true" class="flex" data-popover-trigger=""><div slot="triggerEl" class=""><div class="k-color-picker-block-w k-color-picker-block-w--lg"><div class="k-color-picker-block k-color-picker-block--lg k-color-picker-block--lg"><div class="k-color-picker-block-content--lg" style="background-color: rgb(0, 0, 0);"></div></div> <div slot="text" class="" style="display: none;"></div></div></div></div> "`;
